From 14d307181fd8a31e3e15b0fd2e3eadc1b7632a83 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@nxp.com>
Date: Mon, 22 Jan 2018 14:13:47 +0530
Subject: [PATCH 375/767] mtd: fsl-quadspi: use memcpy_fromio() instead of
 memcpy()

When performing a memcpy opration to_or_from I/O memory, use memcpy_fromio/toio instead of memcpy()

Signed-off-by: Suresh Gupta <suresh.gupta@nxp.com>
Signed-off-by: Abhimanyu Saini <abhimanyu.saini@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 7ecb68789822..1a4ad1750e01 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -943,8 +943,9 @@ static ssize_t fsl_qspi_read(struct spi_nor *nor, loff_t from,
 		len);
 
 	/* Read out the data directly from the AHB buffer.*/
-	memcpy(buf, q->ahb_addr + q->chip_base_addr + from - q->memmap_offs,
-		len);
+	memcpy_fromio(buf,
+		      q->ahb_addr + q->chip_base_addr + from - q->memmap_offs,
+		      len);
 
 	return len;
 }
-- 
2.17.0

